### ISSUE TITLE
Enhance the accessibility of `Select` (`@yamada-ui/select`) component

### ISSUE THREAD

**Issue by koralle** [yamada-ui/yamada-ui#1719]
### 1. Accessibility roles

* [ ] Please refer to the image.

![a11y-roles-tobe-in-select](https://github.com/yamada-ui/yamada-ui/assets/33865215/8c8bcfeb-3bb4-4952-b65d-e9a1f2864570)

### 2. `aria-controls`

* [ ] An element with the `combobox` role must have in its `aria-controls` attribute the ID of the corresponding popup element (the element with the `listbox` role).

### 3. `aria-expanded`

* [ ] When an element with the `listbox` role is hidden, the element with the `combobox` role should have the `aria-expanded` attribute set to `false`.
* [ ] When an element with the `listbox` role is visible, the element with the `combobox` role should have the `aria-expanded ` attribute set to `true`.

### 4. `aria-activedescendant`

* [ ] While a descendant element of the element with the `listbox` role is focused, the element with the `combobox` role must have the ID of the focused element as the value of its `aria-activedescendant` attribute.

### 5. `aria-label` / `aria-labelledby`

* [ ] Please refer to the following flowchart.

```
Does the combobox have a visible label?
    ├──> No: `combobox` element has a label provided by `aria-label` . 
    └──> Yes
           └──> Is the `combobox` element an HTML element that can be labelled using the HTML `label` element (e.g., the input element)?
                  ├──> No: `combobox` element has `aria-labelledby` set to a value that refers to the labelling element. 
                  └──> Yes: `combobox` is labeled using the `label` element. 
```

### 6. Keyboard Interaction

* [ ] When `Alt` + `Up Arrow` is pressed while the popup is displayed...
  * [ ] If an element within the popup is focused, return the focus to the element with the `combobox` role.
  * [ ] Close the popup.

## References

* [Select-Only Combobox Example | APG | WAI | W3C](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-select-only/)
* [Combobox Pattern | APG | WAI | W3C](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/)
* [Select - Yamada UI](https://yamada-ui.com/components/forms/select)
* [Select – React Aria](https://react-spectrum.adobe.com/react-aria/Select.html)

### Problem Statement/Justification

We should prioritize web accessibility to ensure our UI components are usable by everyone, including people with disabilities. This inclusivity enhances user experience and complies with legal standards, broadening Yamada UI's user base.

### Are you willing to participate in implementing this feature and create a pull request include the implementation?

None

**Comment by nakayan5** [comment]
@koralle 
duplicate ?
https://github.com/yamada-ui/yamada-ui/issues/1959

**Comment by koralle** [comment]
@nakayan5 

Have you checked the three implementation examples of the `ListBox` pattern published by W3C APG? 　　
The `Select` component of Yamada UI does not correspond to any of these three examples. 

On the other hand, on [the `ComboBox` pattern page](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/), the combobox widget is clearly defined as follows, and this perfectly fits it. Similar implementation examples are also published.

> A [combobox](https://w3c.github.io/aria/#combobox) is an input widget that has an associated popup. The popup enables users to choose a value for the input from a collection.

**Comment by koralle** [comment]
This is optional.

>  When Alt + Up Arrow is pressed while the popup is displayed...
> * [ ]  If an element within the popup is focused, return the focus to the element with the combobox role.
> * [ ]  Close the popup.
